src/starclans/components/ScanView.tsx

```diff
@@ -1,10 +1,255 @@
-import { Box, Typography } from '@mui/material';
+import React, { useState, useEffect } from 'react';
+import { 
+  Box, 
+  Typography, 
+  Button, 
+  CircularProgress, 
+  Card, 
+  CardContent, 
+  Divider, 
+  List, 
+  ListItem, 
+  ListItemText,
+  LinearProgress,
+  Alert,
+  Paper,
+  Grid
+} from '@mui/material';
+import RadarIcon from '@mui/icons-material/Radar';
+import WarningIcon from '@mui/icons-material/Warning';
+import EmojiEventsIcon from '@mui/icons-material/EmojiEvents';
+import DiamondIcon from '@mui/icons-material/Diamond';
+
+import useStarclanGameStore from '../context/useStarclanGameStore';
+import useStarclanUIStore from '../context/useStarclanUIStore';
+import { ACTION_TYPES, ScanResult } from '../domain/models';
 
 const ScanView = () => {
+  const gameState = useStarclanGameStore((state) => state.gameState);
+  const attemptPlayerAction = useStarclanGameStore((state) => state.attemptPlayerAction);
+  const isApiProcessing = useStarclanUIStore((state) => state.isApiProcessing);
+  const [timeRemaining, setTimeRemaining] = useState<number | null>(null);
+  const [progress, setProgress] = useState<number>(0);
+
+  // Calculate and update time remaining for in-progress scan
+  useEffect(() => {
+    if (!gameState?.activeScan || gameState.activeScan.status !== 'IN_PROGRESS') {
+      setTimeRemaining(null);
+      setProgress(0);
+      return;
+    }
+
+    const startTime = gameState.activeScan.startTime;
+    const endTime = startTime + gameState.activeScan.durationMs;
+    const totalDuration = gameState.activeScan.durationMs;
+
+    const updateTimer = () => {
```

Comment:
not sure I like this updateTimer? seems like we are doing this in other places - maybe there is a pattern we should pull out into a single place?